nc2zarr/writer.py

```diff
@@ -172,3 +182,85 @@ def _remove_variable_attrs(cls, ds: xr.Dataset) -> xr.Dataset:
         for k, v in ds.variables.items():
             v.attrs = dict()
         return ds
+
+    def _finalize_dataset(self):
+        if not self._output_adjust_metadata and not self._output_metadata:
+            return
+
+        with log_duration('Finalizing dataset'):
+            adjusted_metadata = {}
+
+            if self._output_adjust_metadata:
+                self._ensure_store()
+                # Get new attribute values
+                with xr.open_zarr(self._output_store, decode_cf=True) as dataset:
+                    history = self._get_history_metadata(dataset)
+                    source = self._get_source_metadata(dataset)
+                    time_coverage_start, time_coverage_end = \
+                        self._get_time_coverage_metadata(dataset)
+                    adjusted_data = (
+                        ('history', history),
+                        ('source', source),
```

Comment:
I feel that it if there is more than one source we should use the term 'sources'.

Reply:
No, this the CF name for this attribute: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#description-of-file-contents